packages/eslint-plugin/tests/rules/prefer-nullish-coalescing.test.ts

```diff
@@ -333,6 +333,26 @@ x ?? 'foo';
           },
         ],
       },
+      {
+        code: code.replace(/y/g, '(z = y)'),
```

Comment:
[Testing] The changed logic includes `beforeCount` and `afterCount`, but there's only one here. Could you add some tests?

* `((z = y))`
* `( /* */ ( z = y ) /* */ )`
* Maybe, a multiline comment or two?
* ...whatever you think is also a good idea to test

Reply:
Yeah, I'm afraid it doesn't handle any of those cases. the `getTextWithParentheses` function I moved isn't fancy enough to handle that.

Is there a helper that already exists to do this that I'm not aware of? Reprinting a node like this sure seems like a common case.

Reply:
Maybe `getWrappingFixer`? I had looked at that but couldn't reverse engineer how to use it in this application.

Reply:
That would be the one, yeah. Does its use in `strict-boolean-expressions` help?

Reply:
Tried it out, and it adds parens in places it shouldn't, and doesn't preserve multiple parens. So, I don't think I have a good idea...

Reply:
Hmm. Well, I guess you're on your own then. Blurgh.

If there's anything that's an existing bug and would be a lot more work to tackle, I think it's fine as a followup issue. I'll just mark this as ✅ for now.